### PR TITLE
Gallery example "Legend": Update regarding input data and multi-column legends

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

In contrast to the change in PR #2606, it is actually possible to create a multi-column legend for a automatically created legend. This PR corrects the introduction text of the gallery example "Legend".

I feel it would be good to show how to create a horizontal legend, as this is done via the `label` parameter of `Figure.plot`, and not via a parameter of the `Figure.legend` method. For more complicated adjustments of a legend a tutorial seem to make sense. Currently, the same plot is created twice, first with a vertical legend (default) and then with a horizontal legend. However, I am not 100 % happy with this example. I do not like that we plot the same data twice above each other, just with different symbols. And maybe it is not optimal to simply repeat the same plot. Suggestions are welcome 🙂.

Related:
- PyGMT issue: #2603
- PyGMT PR: #2606
- GMT forum: https://forum.generic-mapping-tools.org/t/horizontal-legend/4362

**Preveiw**: https://pygmt-dev--2762.org.readthedocs.build/en/2762/gallery/embellishments/legend.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
> However, I am not 100 % happy with this example. I do not like that we plot the same data twice above each other, just with different symbols. And maybe it is not optimal to simply repeat the same plot.

I agree. Maybe we can simply create some simple sine/cosine curves instead.

**Comment by yvonnefroehlich** [comment]
> > However, I am not 100 % happy with this example. I do not like that we plot the same data twice above each other, just with different symbols. And maybe it is not optimal to simply repeat the same plot.
> 
> I agree. Maybe we can simply create some simple sine/cosine curves instead.

I made some changes regarding the used data (now sine and cosine curves) and the shown legend (now vertical and horizontal legends). I combined this within one figure, a bit similar to the gallery example we have for colorbars. @GenericMappingTools/pygmt-maintainers, what do you think?

File: examples/gallery/embellishments/legend.py [hunk 1]
```diff
@@ -3,32 +3,63 @@
...
+    region=[-10, 10, -3.5, 3.5],
+    frame=["WSne", "xaf+lx", "ya1f0.5+ly"],
 )
-fig.plot(data="@Table_5_11.txt", pen="1.5p,gray", label="My lines")
-fig.plot(data="@Table_5_11.txt", style="t0.40c", fill="orange", label="Oranges")
 
+# -----------------------------------------------------------------------------
+# Top: Vertical legend (one column, default)
+
+# Use the label parameter to state to text label for the legend entry
```
Comment (michaelgrund) [inline examples/gallery/embellishments/legend.py:41]:
```suggestion
# Use the label parameter to state the text label for the legend entry
```

File: examples/gallery/embellishments/legend.py [hunk 2]
```diff
@@ -3,32 +3,63 @@
...
+    frame=["WSne", "xaf+lx", "ya1f0.5+ly"],
 )
-fig.plot(data="@Table_5_11.txt", pen="1.5p,gray", label="My lines")
-fig.plot(data="@Table_5_11.txt", style="t0.40c", fill="orange", label="Oranges")
 
+# -----------------------------------------------------------------------------
+# Top: Vertical legend (one column, default)
+
+# Use the label parameter to state the text label for the legend entry
+fig.plot(x=x, y=y1, pen="1p,green3", label="Sine(x)+1.1")
```
Comment (seisman) [inline examples/gallery/embellishments/legend.py:42]:
```suggestion
fig.plot(x=x, y=y1, pen="1p,green3", label="sin(x)+1.1")
```

File: examples/gallery/embellishments/legend.py [hunk 3]
```diff
@@ -3,32 +3,63 @@
...
+
+fig.plot(x=x, y=y2, style="c0.07c", fill="dodgerblue", label="Cosine(x)+1.1")
```
Comment (seisman) [inline examples/gallery/embellishments/legend.py:44]:
```suggestion
fig.plot(x=x, y=y2, style="c0.07c", fill="dodgerblue", label="cos(x)+1.1")
```

**Review by seisman — APPROVED** [review]
Is this PR still WIP? It looks good to me so approve.

**Comment by yvonnefroehlich** [comment]
> Is this PR still WIP? It looks good to me so approve.

The PR should be finished from my side - updated the title. Just reduced some double blank lines.

@seisman I am sorry for writing in your comment 🙁 - I reverted this change.

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**